### ISSUE TITLE
Remove dependency from dirgraph format

### ISSUE THREAD

**Issue by sulmar** [dotnet-state-machine/stateless#557]
Hi,
I'm working on exporting the state machine to the **mermaid format**.

I have currently like this:

``` csharp
public class MermaidGraphStyle : GraphStyleBase
{
    public override string GetPrefix()
    {
        return "stateDiagram-v2";
    }

  // ...

``` 

and unit test:

``` csharp
// Arrange
StateMachine<string, string> _machine = new StateMachine<string, string>("s1");

_machine.Configure("s1")
    .Permit("A transition", "s2");

var machineInfo = _machine.GetInfo();

// Act
var result = Stateless.Graph.MermaidGraph.Format(machineInfo);

// Assert
string expected = "stateDiagram-v2
	s1 --> s2 : A transition";

Assert.AreEqual(expected, result);
 
```

In result I get:

``` 
stateDiagram-v2
	s1 --> s2 : A transition
 init [label="", shape=point];
 init -> "s1"[style = "solid"]
}
```

The problem is because in _ToGraph()_ method there is a strong dependence on the digraph format:

 ``` csharp
public string ToGraph(GraphStyleBase style)
{
 ...
 // Add initial transition if present
 var initialStateName = initialState.UnderlyingState.ToString();
 dirgraphText += System.Environment.NewLine + $" init [label=\"\", shape=point];";
 dirgraphText += System.Environment.NewLine + $" init -> \"{initialStateName}\"[style = \"solid\"]";

 dirgraphText += System.Environment.NewLine + "}";

...
}

```

Could you please remove in Stata Graph.To Graph() method dependency from dirgraph format?

Maybe good idea is refactor this section into abstract method in GraphStyleBase class?

**Comment by sulmar** [comment]
I just created pull request #558

**Comment by mclift** [comment]
Resolved in release 5.15.0
